### PR TITLE
[WIP] Refactor test helpers

### PR DESCRIPTION
**Pull request by dannyd4315**
Closes https://github.com/cookpad/streamy/issues/70 - Refactored the test helpers so that both minitest and rspec helpers function the same way. Also move some spec around to ensure they are in the correct spec class. 

I am also proposing to move some of the spec helpers such as `expect_no_event` to streamy in another PR. - See [here for helpers in `global-web`](https://github.com/cookpad/global-web/blob/6fd50d3b7d600b36dbd20720af003aa3a04e9a8e/spec/support/helpers/event_stream_helper.rb)

I will also update the documentation on spec helpers in a separate PR.

### CODE REVIEW

File: lib/streamy/helpers/message_parser.rb [hunk 1]
```diff
@@ -3,20 +3,18 @@
...
 
-      def parse_message(message_payload, encoding_type)
-        if encoding_type == :avro
-          avro.decode(message_payload).deep_symbolize_keys
-        else
-          JSON.parse(message_payload).deep_symbolize_keys
-        end
+      def parse_message(message_payload)
+        JSON.parse(message_payload).deep_symbolize_keys
+      rescue JSON::ParserError
```
Comment (ChrisBr) [inline lib/streamy/helpers/message_parser.rb]: Wondering what happens if the payload is neither valid json nor avro? Do we need to have another rescue? 

Also is there no other way to know if the payload is avro or json? Is it not included in the message itself?

Comment (dannyd4315) [reply lib/streamy/helpers/message_parser.rb]: Yeah this is one of those [`cards on the table`](https://github.com/cookpad/web-chapter/pull/115#event-2284436509) moments I suppose. I had been thinking about this somewhat and couldn't come up with a cleaner solution. The `Streamy.message_bus` would have two different types of events on (avro and json), and we need the specs to iterate over the deliveries in that message bus and decode them to a hash format. 

You would not be able to encode into an avro format if the attributes were incorrect, and its the same with json, you would get an error when the `JsonEvent` coverted the payload to json. So in my mind we are are covered by the input errors that you would get on publishing. Meaning that the message payload could only be either valid json or valid avro.

Comment (dannyd4315) [reply lib/streamy/helpers/message_parser.rb]: Another approach could be to create a new message bus for each event encoding type within the specs. I.e a `Streamy.message_bus` for avro encoded events and another `Streamy.message_bus` for json encoded events

**Comment by dannyd4315** [comment]
After much deliberation I decided to add a new attribute to each message of `encoding_format:` the benefits of this are two fold, we/or other developers can then create different encoding classes `protobuf` for example and put messages with different encoding types on the same topics. These can then be consumed by the consumer which would know know how the payload is encoded. 

This I think simplifies the implementation and is a cleaner solution. This should not effect any of our current producers or consumers. 

Any thoughts on this implementation @balvig? Both the minitest/rspec helpers also now function in the same way.

File: lib/streamy/event.rb [hunk 2]
```diff
@@ -62,6 +63,10 @@ def event_time
         raise "event_time must be implemented on #{self.class}"
       end
 
+      def encoding_format
+        raise "encoding_format must be implemented on the encoding class"
```
Comment (lewispb) [inline lib/streamy/event.rb]:
```suggestion
        raise "encoding_format must be implemented on #{self.class}"
```

File: lib/streamy/helpers/message_parser.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@ module MessageParser
       def hashify_messages(message_bus_deliveries)
         message_bus_deliveries.map do |message|
           message_hash = message.dup
-          message_hash[:payload] = parse_message(message_hash[:payload], message_hash[:encoding_format])
+          message_hash[:payload] = parse_message(message_hash[:payload], message_hash[:headers][:encoding_format])
```
Comment (ChrisBr) [inline lib/streamy/helpers/message_parser.rb]: Can `:headers` ever be nil?

Or should we use ``message_hash.dig(:headers, :encoding_format)`` instead?

**Pull request closed without merging**